webapp/components/generic/SearchableInput/SearchableInput.vue

```diff
@@ -89,9 +89,18 @@ export default {
     },
     handleInput(event) {
       clearTimeout(this.searchProcess)
-      this.value = event.target ? event.target.value.replace(/\s+/g, ' ').trim() : ''
+      this.value = event.target ? event.target.value.replace(/[-\s]+/g, ' ').trim() : ''
       this.unprocessedSearchInput = this.value
-      if (isEmpty(this.value) || this.value.replace(/\s+/g, '').length < 3) {
+      if (isEmpty(this.value)) {
+        return
+      }
+      if (this.value.includes(' ')) {
+        this.value = this.value
+          .split(' ')
+          .filter(word => word.length > 2)
+          .join(' ')
+      }
+      if (this.value.length < 3) {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-07T11:04:58Z" title="Friday, February 7th 2020, 12:04:58 pm +01:00">Feb 7, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -89,9 +89,18 @@ export default {
     },
     handleInput(event) {
       clearTimeout(this.searchProcess)
-      this.value = event.target ? event.target.value.replace(/\s+/g, ' ').trim() : ''
+      this.value = event.target ? event.target.value.replace(/[-\s]+/g, ' ').trim() : ''
       this.unprocessedSearchInput = this.value
-      if (isEmpty(this.value) || this.value.replace(/\s+/g, '').length < 3) {
+      if (isEmpty(this.value)) {
+        return
+      }
+      if (this.value.includes(' ')) {
+        this.value = this.value
+          .split(' ')
+          .filter(word => word.length > 2)
+          .join(' ')
+      }
+      if (this.value.length < 3) {
```


This is a quite hacky solution. I'm going to approve this but I feel bad about it. Do you know "stopwords"? It's a feature of search indices and I think that would have been the proper solution.

@Mogge if you feel like it, you could add an integration test for it to document this behaviour at least.